selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/NewWorkspacePageTest.java

```diff
@@ -0,0 +1,519 @@
+    newWorkspace.waitSelectedFiltersSuggestion("JAVA 1.8, THEIA");
+
+    newWorkspace.singleClickOnFiltersSuggestions("JAVA");
+    newWorkspace.waitSelectedFiltersSuggestion("JAVA");
+
+    newWorkspace.singleClickOnFiltersSuggestions("JDK");
+    newWorkspace.waitSelectedFiltersSuggestion("JDK");
+
+    newWorkspace.doubleClickOnFiltersSuggestion("JDK");
+    newWorkspace.waitFiltersInputFieldTags(asList("JDK"));
+
+    newWorkspace.deleteLastTagFromInputTagsField();
+    newWorkspace.waitFiltersInputFieldIsEmpty();
+
+    newWorkspace.typeToFiltersInput("j");
+    newWorkspace.waitFiltersSuggestionsNames(EXPECTED_FILTERS_SUGGESTIONS);
+
+    newWorkspace.waitSelectedFiltersSuggestion("JAVA");
+    newWorkspace.doubleClickOnFiltersSuggestion("JAVA 1.8, THEIA");
+
+    newWorkspace.waitFiltersInputFieldTags(asList("JAVA 1.8, THEIA"));
+
+    newWorkspace.deleteTagByRemoveButton("JAVA 1.8, THEIA");
+    newWorkspace.waitFiltersInputFieldIsEmpty();
+
+    newWorkspace.typeToFiltersInput("j");
+    newWorkspace.waitFiltersSuggestionsNames(EXPECTED_FILTERS_SUGGESTIONS);
+    newWorkspace.waitSelectedFiltersSuggestion("JAVA");
+    newWorkspace.chooseFilterSuggestionByPlusButton("JDK");
+    newWorkspace.waitFiltersInputFieldTags(asList("JDK"));
+    newWorkspace.clickOnInputFieldTag("JDK");
+    seleniumWebDriverHelper.sendKeys(Keys.DELETE.toString());
+    newWorkspace.waitFiltersInputFieldIsEmpty();
+
+    newWorkspace.typeToFiltersInput("j");
+    newWorkspace.waitFiltersSuggestionsNames(EXPECTED_FILTERS_SUGGESTIONS);
+    newWorkspace.waitSelectedFiltersSuggestion("JAVA");
+    newWorkspace.chooseFilterSuggestionByPlusButton("JAVA");
+    newWorkspace.waitFiltersInputFieldTags(asList("JAVA"));
+    newWorkspace.clickOnInputFieldTag("JAVA");
+    seleniumWebDriverHelper.sendKeys(Keys.DELETE.toString());
+    newWorkspace.waitFiltersInputFieldIsEmpty();
+    newWorkspace.deleteLastTagFromInputTagsField();
+
+    newWorkspace.typeToFiltersInput("j");
+    newWorkspace.waitSelectedFiltersSuggestion("JAVA");
+    seleniumWebDriverHelper.sendKeys(Keys.TAB.toString());
+    newWorkspace.waitSelectedFiltersSuggestion("JDK");
+    seleniumWebDriverHelper.sendKeys(Keys.ENTER.toString());
+    newWorkspace.waitFiltersInputFieldTags(asList("JDK"));
+    newWorkspace.clickOnTitlePlaceCoordinate();
+    newWorkspace.waitFiltersFormClosed();
+
+    newWorkspace.getAvailableStacks();
+    newWorkspace.waitStacks(EXPECTED_JDK_STACKS);
+
+    newWorkspace.clickOnFiltersButton();
+    newWorkspace.waitFiltersFormOpened();
+    newWorkspace.waitFiltersInputFieldTags(asList("JDK"));
+    newWorkspace.deleteLastTagFromInputTagsField();
+    newWorkspace.waitFiltersInputFieldIsEmpty();
+    newWorkspace.clickOnTitlePlaceCoordinate();
+    newWorkspace.waitFiltersFormClosed();
+    newWorkspace.waitStacks(EXPECTED_QUICK_START_STACKS);
+  }
+
+  @Test
+  public void checkAddStackButton() {
+    newWorkspace.waitPageLoad();
```

Comment:
There is a list similar type of steps here and it's hard to understand how they are differ from each other. So, it would be better to have different steps commented, or extracted into the separate tests.